### PR TITLE
`caseCon` and the evaluator recognise `errorX` as bottom

### LINKED ISSUE #1507: `autoReg @System @(Maybe Int)` can't be used with undefined reset values
Minimal example:

```haskell
import Clash.Prelude

topEntity :: 
  SystemClockResetEnable => 
  Signal System (Maybe Int) ->
  Signal System (Maybe Int)
topEntity = autoReg  (errorX "foo bar")       
```

Error:

```
Test.hs:7:1: error:
    
    Clash.Netlist.BlackBox(188): Couldn't instantiate blackbox for Clash.Signal.Internal.register#. Verification procedure reported:
    
    Argument 5 should be literal, as blackbox used ~CONST[5], but was:
    
    Identifier "initial_0" Nothing
    
    The source location of the error is not exact, only indicative, as it is acquired 
    after optimizations. The actual location of the error can be in a function that is 
    inlined. To prevent inlining of those functions, annotate them with a NOINLINE pragma.
  |
7 | topEntity = autoReg  (errorX "foo bar")
  | ^^^^^^^^^
```

When running with `-fclash-debug DebugSilent` I get:

```
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Irreducible constant as case subject: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
Can be reduced to: Clash.XException.errorX
  @(GHC.Maybe.Maybe[3674937295934324792] GHC.Types.Int[3674937295934324766])
  (Clash.Transformations.removedArg @GHC.Stack.Types.CallStack[3674937295934325108])
  (Clash.Transformations.removedArg
     @(GHC.Types.[][3674937295934324788] GHC.Types.Char[3674937295934324752]))
```

### PR DESCRIPTION
**Pull request by christiaanb**
Fixes #1507

### CODE REVIEW

File: clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs [hunk 1]
```diff
@@ -103,7 +103,8 @@ import {-# SOURCE #-} Clash.GHC.Evaluator
 
 isUndefinedPrimVal :: Value -> Bool
 isUndefinedPrimVal (PrimVal (PrimInfo{primName}) _ _) =
-  primName == "Clash.Transformations.undefined"
+  primName `elem` ["Clash.Transformations.undefined"
+                  ,"Clash.XException.errorX"]
```
Comment (martijnbastiaan) [inline clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs:107]: 1. Can the evaluator access blackbox internals? If so, could we replicate the behavior in `Clash.Normalize.Transformations.isPrimError`? 

2. Should we add `"Clash.Transformations.removedArg"` too?

**Review by martijnbastiaan — COMMENTED** [review]
Couple of questions, depending on the answer changes requested.

Could you add a CHANGELOG?

File: clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs (hunk 1 above)
Comment (christiaanb) [reply clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs:107]: 1. No, it does not have access
2. No, definitely not... having `removedArg` be a subject of a case-expression would be a bug

**Comment by christiaanb** [comment]
Added the CHANGELOG entry

**Review by martijnbastiaan — APPROVED** [review]

**Pull request merged**